xbmc/cores/RetroPlayer/rendering/RenderContext.cpp

```diff
@@ -70,16 +62,16 @@ void CRenderContext::ApplyStateBlock()
   m_rendering->ApplyStateBlock();
 }
 
-void CRenderContext::EnableGUIShader()
+void CRenderContext::EnableGUIShader(GL_SHADER_METHOD method)
 {
 #if defined(HAS_GL)
   CRenderSystemGL *rendering = dynamic_cast<CRenderSystemGL*>(m_rendering);
   if (rendering != nullptr)
-    rendering->EnableShader(SM_TEXTURE);
+    rendering->EnableShader(ESHADERMETHOD(method));
```

Comment:
Use a translator instead of casting, as there's no guarantee enums will be in the same order or have the same values.

Normally a switch statement would suffice, but because `method` is used twice, you should create a translator function in an anonymous namespace directly before the constructor:

```c++
#if defined(HAS_GL) || defined(HAS_GLES)
namespace
{
ESHADERMETHOD TranslateShaderMethod(GL_SHADER_METHOD method)
{
  switch (method)
  {
  case ... return ...;
  case ... return ...;
  default:
    break;
  }

  return GL_SHADER_METHOD::DEFAULT:
}
}
#endif
```